File: frontend/src/app/features/tenants/models/tenant.model.ts
import { PaginationQuery } from '../../../core/models';

export interface Tenant {
  id: string;
  organizationId: string;
  userId?: string | null;
  fullName: string;
  phone: string;
  email?: string | null;
  profession?: string | null;
  employer?: string | null;
  monthlyIncome?: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTenantDto {
  fullName: string;
  phone: string;
  email?: string;
  profession?: string;
  employer?: string;
  monthlyIncome?: number;
}

export type UpdateTenantDto = Partial<CreateTenantDto>;

export interface TenantQuery extends PaginationQuery {
  search?: string;
}

File: mobile/lib/core/models/tenant_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'tenant_model.freezed.dart';
part 'tenant_model.g.dart';

/// A tenant record (as distinct from a `LOCATAIRE` portal user account) -
/// `GET /tenants`, `GET /tenants/:id`, `POST /tenants`
/// (`TenantsMapper.toResponse`). `userId` is only set once the tenant has
/// activated their portal account via a `TenantInvitation`.
@freezed
class TenantModel with _$TenantModel {
  const factory TenantModel({
    required String id,
    required String organizationId,
    String? userId,
    required String fullName,
    required String phone,
    String? email,
    String? profession,
    String? employer,
    num? monthlyIncome,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _TenantModel;

  factory TenantModel.fromJson(Map<String, dynamic> json) =>
      _$TenantModelFromJson(json);
}
